read_until/base.py

```diff
@@ -252,12 +267,11 @@ def __init__(self, mk_host='127.0.0.1', mk_port=8000, cache_size=512, cache_type
 
         self.grpc_port = self.mk_grpc_port
         self.logger.info('Creating rpc connection on port {}.'.format(self.grpc_port))
-        self.connection = minknow.rpc.Connection(host=self.mk_host, port=self.grpc_port)
+        self.connection = minknow_api.Connection(host=self.mk_host, port=self.grpc_port)
         self.logger.info('Got rpc connection.')
         self.msgs = self.connection.data._pb
-        self.device = minknow.Device(self.connection)
 
-        self.signal_dtype = self.device.numpy_data_types.calibrated_signal
+        self.signal_dtype = _numpy_type(self.connection.data.get_data_types().calibrated_signal)
```

Comment:
you should be able to use `minknow_api.get_numpy_types(self.connection).calibrated_signal`, rather than copying in `_numpy_type`

Reply:
Done